src/fixtures/primitives.ts

```diff
@@ -65,25 +65,41 @@ export const blsSignatureBytes = () =>
 export const blsSignature = () =>
   BlsSignature.fromBytes(blsSignatureBytes())[0];
 
+/**
+ * Example WarpMessage bytes.
+ *
+ * WarpUnsignedMessage:
+ * - networkId: 12345
+ * - sourceChainId: rVPmsy4tQrQhQpRE6embdYnmYtwQi8kQAg1ZdaZGzeVdjc4qT
+ *
+ * One "signer" on the WarpSignature.
+ */
 export const warpMessageBytes = () =>
   new Uint8Array([
-    0x00, 0x00, 0x00, 0xd7, 0x00, 0x00, 0x00, 0x00, 0x30, 0x39, 0x70, 0x5f,
-    0x3d, 0x44, 0x15, 0xf9, 0x90, 0x22, 0x5d, 0x3d, 0xf5, 0xce, 0x43, 0x7d,
-    0x7a, 0xf2, 0xaa, 0x32, 0x4b, 0x1b, 0xbc, 0xe8, 0x54, 0xee, 0x34, 0xab,
-    0x6f, 0x39, 0x88, 0x22, 0x50, 0xd2, 0x00, 0x00, 0x00, 0x44, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x36,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x38, 0xe6, 0xe9, 0xfe, 0x31, 0xc6,
-    0xd0, 0x70, 0xa8, 0xc7, 0x92, 0xdb, 0xac, 0xf6, 0xd0, 0xae, 0xfb, 0x8e,
-    0xac, 0x2a, 0xde, 0xd4, 0x9c, 0xc0, 0xaa, 0x9f, 0x42, 0x2d, 0x1f, 0xdd,
-    0x9e, 0xcd, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x05, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x01, 0x01, 0x87, 0xf4, 0xbb, 0x2c, 0x42, 0x86, 0x9c, 0x56, 0xf0,
-    0x23, 0xa1, 0xca, 0x81, 0x04, 0x5a, 0xff, 0x03, 0x4a, 0xcd, 0x49, 0x0b,
-    0x8f, 0x15, 0xb5, 0x06, 0x90, 0x25, 0xf9, 0x82, 0xe6, 0x05, 0xe0, 0x77,
-    0x00, 0x7f, 0xc5, 0x88, 0xf7, 0xd5, 0x63, 0x69, 0xa6, 0x5d, 0xf7, 0x57,
-    0x4d, 0xf3, 0xb7, 0x0f, 0xf0, 0x28, 0xea, 0x17, 0x37, 0x39, 0xc7, 0x89,
-    0x52, 0x5a, 0xb7, 0xee, 0xbf, 0xcb, 0x5c, 0x11, 0x5b, 0x13, 0xcc, 0xa8,
-    0xf0, 0x2b, 0x36, 0x21, 0x04, 0xb7, 0x00, 0xc7, 0x5b, 0xc9, 0x52, 0x34,
-    0x10, 0x9f, 0x3f, 0x13, 0x60, 0xdd, 0xcb, 0x4e, 0xc3, 0xca, 0xf6, 0xb0,
-    0xe8, 0x21, 0xcb,
+    0x00, 0x00, 0x00, 0x00, 0x30, 0x39, 0x70, 0x5f, 0x3d, 0x44, 0x15, 0xf9,
```

Comment:
@meeh0w did you supply this warp message? Where did you get it from? Is it well formed, do you think it might contain some `0x` prefix which might explain the START_INDEX referenced below?  https://github.com/ava-labs/avalanchejs/pull/919/files#diff-152fa1d7798a9c503553385990a132442d7b70513c85848252496bc011cc60feR8

Reply:
This is the one I used for `RegisterL1Validator`:

```
00000000004c4f0bad9d4bc36b7d75e6d5d579a10d5309e23f04e8e05ab727015ffd44a61e39000000d8000000000001000000145f584c2d56b4c356e7d82ec6129349393dc5df17000000b600000000000132704397a7b04615df4fc7e8f1188e00541d4dfd081ccc29027623cc534019d00000001404b3c85c01c4cc96a9cb8e6a8f3387f177370855ab9359160057de14fc9c33ff4277c56927954f10e0c53c5fe60b8c68688e33a28f4184eeb6ce4717ea9176e43671e4f400000000673c4fbb00000001000000013cb7d3842e8cee6a0ebd09f1fe884f6861e1b29c00000001000000013cb7d3842e8cee6a0ebd09f1fe884f6861e1b29c000000000000001400000000000000011ba27f305b330b4cf355f676436b9c7aefa2656618b5da9409001561202750fb510f4915aeaae3f1d050ab76485ea240d416513061ef1bfdffcdc78ab0beda27f9d4dc6e8d4281c93702d68e9e8422b30ce13c41b75ed14610b5c98546448ad4c3
```

I got it from the Avalanche CLI, `blockchain addValidator` command with `CallAddValidator` function modified so it does not dispatch the tx, but instead exits right after getting the signed warp message:

<img width="1390" alt="image" src="https://github.com/user-attachments/assets/6fb03024-c5e8-469f-887e-f9210470656a">

Reply:
The exact order of commands I've been using is:

```
// create subnet
./bin/avalanche blockchain create <name> --evm --proof-of-authority

// create chain + convert to L1
// make sure you have AvaGo built on the latest release
./bin/avalanche blockchain deploy <name> --etna-devnet --use-local-machine --avalanchego-path=/Users/meehow/ava/avalanchego/build/avalanchego

// register the validator (replace <name> with the blockchain name, in the cluster param as well)
./bin/avalanche blockchain addValidator <name> --ewoq --cluster <name>-local-node
```

Reply:
Tested against the AvalancheGo messages and this message above and everything seems to work as expected with the latest changes (ie proper serializers/deserializers).

The first bytes are the codec version. The other bytes skipped over before the bls signature is the type id deserialized item.

So again, the new codec serializers/deserializers handle all this now.